findmyparadise/src/main/java/com/tp/wrc/findmyparadise/services/ListingServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.tp.wrc.findmyparadise.services;
+
+import com.tp.wrc.findmyparadise.models.Listing;
+import com.tp.wrc.findmyparadise.repositories.ListingRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ListingServiceImpl implements ListingService {
+
+    @Autowired
+    private ListingRepository repo;
+
+    @Override
+    public List<Listing> index() {
+        return repo.findAll();
+    }
+
+    @Override
+    public Listing show(int id) {
+        Optional<Listing> listing = repo.findById(id);
```

Comment:
Optional has an OrElseThrow method that could be useful for throwing an exception here instead of returning null.